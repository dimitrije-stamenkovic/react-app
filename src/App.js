import React, { useState } from 'react';

function App() {

  const [count, setCount] = useState(0);

  return (
    <div className="App bg-slate-200">

      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 flex justify-center items-center h-screen">
        {/* We've used 3xl here, but feel free to try other max-widths based on your needs */}
        <div className="max-w-3xl mx-auto ">

          <p data-testid="counter" className='my-8 text-2xl text-center'>{count}</p>

          <div class="flex">

            <button
              type="button"
              data-testid="button-1"
              onClick={() => setCount(count + 1)}
              className="mx-1 inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            >
              +1
            </button>

            <button
              type="button"
              data-testid="button-2"
              onClick={() => setCount(count + 2)}
              className="mx-1 inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            >
              +2
            </button>

            <button
              type="button"
              data-testid="button-3"
              onClick={() => setCount(count - 1)}
              className="mx-1 inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            >
              -1
            </button>

            <button
              type="button"
              data-testid="button-4"
              onClick={() => setCount(count - 2)}
              className="mx-1 inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            >
              -2
            </button>

          </div>


        </div>

      </div>
    </div>
  );
}

export default App;
