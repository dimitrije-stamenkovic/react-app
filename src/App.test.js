import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect";
import App from './App'

afterEach(cleanup)

it('should take a snapshot', () => {
    const { asFragment } = render(<App />)

    expect(asFragment(<App />)).toMatchSnapshot()

});


it('incerment 1', () => {
    const { getByTestId } = render(<App />);

    fireEvent.click(getByTestId('button-1'))

    expect(getByTestId('counter')).toHaveTextContent('1')
});

it('incerment 2', () => {
    const { getByTestId } = render(<App />);

    fireEvent.click(getByTestId('button-2'))

    expect(getByTestId('counter')).toHaveTextContent('2')
});

it('decrement 1', () => {
    const { getByTestId } = render(<App />);

    fireEvent.click(getByTestId('button-3'))

    expect(getByTestId('counter')).toHaveTextContent('-1')
});

it('decrement 2', () => {
    const { getByTestId } = render(<App />);

    fireEvent.click(getByTestId('button-4'))

    expect(getByTestId('counter')).toHaveTextContent('-2')
});

