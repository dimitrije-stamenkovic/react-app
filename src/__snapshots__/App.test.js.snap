// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App bg-slate-200"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 flex justify-center items-center h-screen"
    >
      <div
        class="max-w-3xl mx-auto "
      >
        <p
          class="my-8 text-2xl text-center"
          data-testid="counter"
        >
          0
        </p>
        <div
          class="flex"
        >
          <button
            class="mx-1 inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            data-testid="button-1"
            type="button"
          >
            +1
          </button>
          <button
            class="mx-1 inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            data-testid="button-2"
            type="button"
          >
            +2
          </button>
          <button
            class="mx-1 inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            data-testid="button-3"
            type="button"
          >
            -1
          </button>
          <button
            class="mx-1 inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            data-testid="button-4"
            type="button"
          >
            -2
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
